src/openms/source/ANALYSIS/OPENSWATH/OpenSwathScoring.cpp

```diff
@@ -627,69 +669,79 @@ namespace OpenMS
       closest_idx--;
     }
 
-    if (nr_spectra_to_add == 1)
+    all_spectra.push_back(swath_map->getSpectrumById(closest_idx));
+    for (int i = 1; i <= nr_spectra_to_fetch / 2; i++) // cast to int is intended!
     {
-      added_spec = swath_map->getSpectrumById(closest_idx);
-      if (drift_upper > 0) 
+      if (closest_idx - i >= 0)
       {
-        added_spec = filterByDrift(added_spec, drift_lower, drift_upper);
+        all_spectra.push_back(swath_map->getSpectrumById(closest_idx - i));
+      }
+      if (closest_idx + i < (int)swath_map->getNrSpectra())
+      {
+        all_spectra.push_back(swath_map->getSpectrumById(closest_idx + i));
       }
     }
-    else
+
+    return all_spectra;
+  }
+
+  OpenSwath::SpectrumPtr OpenSwathScoring::getAddedSpectra_(std::vector<OpenSwath::SpectrumPtr>& all_spectra, const double drift_lower, const double drift_upper)
+  {
+    OpenSwath::SpectrumPtr added_spec(new OpenSwath::Spectrum);
+    added_spec->getDataArrays().push_back( OpenSwath::BinaryDataArrayPtr(new OpenSwath::BinaryDataArray) );
+    added_spec->getDataArrays().back()->description = "Ion Mobility";
```

Comment:
magic keyword.
How do you know? :)
maybe https://github.com/OpenMS/OpenMS/blob/develop/src/openswathalgo/include/OpenMS/OPENSWATHALGO/DATAACCESS/DataStructures.h#L270?
But a common global var would be much safer, see https://github.com/OpenMS/OpenMS/blob/develop/src/openswathalgo/include/OpenMS/OPENSWATHALGO/DATAACCESS/DataStructures.h#L270t

Reply:
Both of these links seem to be the same thing?

Reply:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/CONCEPT/Constants.h#L266 :)